.ci/load/Jenkinsfile

```diff
@@ -0,0 +1,169 @@
+        // End script auto-generation
+    }
+
+    stages {
+        stage('Pre-flight'){
+            steps {
+                echo 'Getting authentication information from Vault'
+                withSecretVault(secret: 'secret/apm-team/ci/bandstand', user_var_name: 'APP_TOKEN_TYPE', pass_var_name: 'APP_TOKEN'){
+                    setEnvVar('SESSION_TOKEN', sh(script: ".ci/load/scripts/start.sh", returnStdout: true).trim())
+                }
+            }    
+        }
+        stage('Load test') {
+            parallel {
+                stage('Load generation') {
+                    agent { label 'metal' }
+                    steps {
+                            withSecretVault(secret: 'secret/apm-team/ci/bandstand', user_var_name: 'APP_TOKEN_TYPE', pass_var_name: 'APP_TOKEN'){
+                                echo 'Preparing load generation..'
+                                whenTrue(Boolean.valueOf(params.locustfile)) {
+                                    echo 'Using user-supplied plan for load-generation with Locust'
+                                    sh script: "echo \"${params.locustfile}\">.ci/load/scripts/locustfile.py"
+                                }
+                                sh(script: ".ci/load/scripts/load_agent.sh")
+                            }
+                    }
+                }
+                stage('Test application') {
+                    agent { label 'metal' }
+                    stages{
+                        stage('Provision Java') {
+                            steps {
+                                echo "Provisioning Java version: ${params.jvm_version}"
+                                setEnvVar('JAVA_HOME', sh(script: ".ci/load/scripts/fetch_sdk.sh ${params.jvm_version}", returnStdout: true).trim())
+                                setEnvVar('JAVACMD', "${env.JAVA_HOME}/bin/java")
+                                setEnvVar('PATH', "${env.JAVA_HOME}/bin:$PATH")
+                            }
+                        }
+                        stage ('Provision agent') {
+                            steps {
+                                echo 'Checking out master branch'
+                                dir("${AGENT_BASE_DIR}") {
+                                    gitCheckout(
+                                        basedir: "apm-agent-java",
+                                        branch: 'master',
+                                        repo: "https://github.com/elastic/${REPO}.git",
+                                        credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+                                        shallow: false
+                                    )
+                                    dir("apm-agent-java"){
+                                        echo 'Switching to requested version'
+                                        sh(script: "git checkout v${params.apm_version}")
```

Comment:
a likely feature request in the future will be to  checkout specific revisions or master

Reply:
Aye. I bumped that (in my head) to an upcoming iteration but it should be do-able, no problem.

Reply:
(Very delayed follow-up) It looks like this is [now handled and that hashes can be passed in](https://github.com/elastic/apm-agent-java/blob/master/.ci/load/Jenkinsfile#L31).